ai21/clients/aws_http_client/aws_http_client.py

```diff
@@ -0,0 +1,240 @@
+import json
+from abc import ABC
+from functools import lru_cache
+from typing import Dict, Optional, TypeVar, Union, Any
+
+import boto3
+import httpx
+
+
+from ai21 import AI21APIError
+from ai21.clients.aws_http_client.aws_authorization import AWSAuthorization
+from ai21.errors import AccessDenied, NotFound, APITimeoutError, ModelStreamError
+from ai21.http_client.async_http_client import AsyncHttpClient
+from ai21.http_client.http_client import HttpClient
+
+_HttpClientT = TypeVar("_HttpClientT", bound=Union[HttpClient, AsyncHttpClient])
+
+
+DEFAULT_AWS_REGION = "us-east-1"
+
+
+class BaseAWSHttpClient(ABC):
```

Comment:
I was actually thinking about this. This class might not be necessary at all and you could use the regular http client that we have. let's talk about this